### ISSUE TITLE
ArrayIndexOutOfBoundsException when moving caches

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9048]
**Describe the bug:**
Crash when moving all caches from a list while in selection mode

**To Reproduce:**
- Open list of saved caches
- Enter selection mode
- Select all caches of this list
- Select Menu - Manage caches - Move selected
- Move caches to a new list
- Answer prompt to delete current list with "Yes"

**Actual behavior/state after performing these steps:**
Crash

**Expected behavior/state after performing these steps:**
No crash

**Version of c:geo used:**
2020.09.23-NB but also reported for 2020.08.04

**Is the problem reproducible:**
Yes

**System information:**
```
--- System information ---
Device: SM-G980F (x1seea, samsung)
Android version: 10
Android build: QP1A.190711.020.G980FXXS4BTHH
c:geo version: 2020.09.23-NB-3f87172
Google Play services: disabled - 20.33.15 (120408-330018294)
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: pres
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide caches: own/found
Hide waypoints: original parking
HW acceleration: enabled (default state)
System language: de_DE
System date format: dd.MM.yy
Debug mode active: no
System internal c:geo dir: /data/user/0/cgeo.geocaching (71,5 GB free) internal
User storage c:geo dir: /storage/emulated/0/cgeo (71,5 GB free) external non-removable
Geocache data: /storage/F8C9-8968/Android/data/cgeo.geocaching/files/GeocacheData (22,2 GB free) external removable
Database: /data/user/0/cgeo.geocaching/databases/data (22,6 MB) on system internal storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   extremcaching.com: Logged in (Anmeldung OK)
   opencaching.de: Logged in (Anmeldung OK)
Geocaching.com date format: d.MM.yyyy
Installed c:geo plugins:  none
BRouter connection available: false
--- End of system information ---
```

**Additional context:**
Stacktrace:

```
09-23 20:24:23.881 21770 21770 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=0; index=0
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at cgeo.geocaching.ui.CacheListAdapter.getPositionForSection(CacheListAdapter.java:771)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FastScroller.getPosFromItemCount(FastScroller.java:1302)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FastScroller.onItemCountChanged(FastScroller.java:534)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.AbsListView.onLayout(AbsListView.java:3054)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at androidx.appcompat.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:530)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:1099)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.View.layout(View.java:23753)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:7277)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3678)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3138)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2199)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9088)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:999)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:797)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:732)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:984)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8154)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
09-23 20:24:23.881 21770 21770 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

**Comment by Lineflyer** [comment]
Still happening on our current beta version 2020.11.14-RC
Seems to be the crash with highest occurrence (21 users) on this beta. I think its not occurring more often with this beta, but it seems to be the last major crash reason for now (hopefully).

Any ideas if/how to fix it before releasing?

**Comment by Lineflyer** [comment]
Stacktrace from 2020.11.14-RC:

```
java.lang.ArrayIndexOutOfBoundsException: 
  at cgeo.geocaching.ui.CacheListAdapter.getPositionForSection (CacheListAdapter.java:771)
  at android.widget.FastScroller.getPosFromItemCount (FastScroller.java:1302)
  at android.widget.FastScroller.onItemCountChanged (FastScroller.java:534)
  at android.widget.AbsListView.onLayout (AbsListView.java:3054)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.widget.LinearLayout.setChildFrame (LinearLayout.java:1829)
  at android.widget.LinearLayout.layoutVertical (LinearLayout.java:1673)
  at android.widget.LinearLayout.onLayout (LinearLayout.java:1582)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.widget.FrameLayout.layoutChildren (FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout (FrameLayout.java:270)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at androidx.appcompat.widget.ActionBarOverlayLayout.onLayout (ActionBarOverlayLayout.java:530)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.widget.FrameLayout.layoutChildren (FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout (FrameLayout.java:270)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.widget.LinearLayout.setChildFrame (LinearLayout.java:1829)
  at android.widget.LinearLayout.layoutVertical (LinearLayout.java:1673)
  at android.widget.LinearLayout.onLayout (LinearLayout.java:1582)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.widget.FrameLayout.layoutChildren (FrameLayout.java:332)
  at android.widget.FrameLayout.onLayout (FrameLayout.java:270)
  at com.android.internal.policy.DecorView.onLayout (DecorView.java:1099)
  at android.view.View.layout (View.java:23753)
  at android.view.ViewGroup.layout (ViewGroup.java:7277)
  at android.view.ViewRootImpl.performLayout (ViewRootImpl.java:3678)
  at android.view.ViewRootImpl.performTraversals (ViewRootImpl.java:3138)
  at android.view.ViewRootImpl.doTraversal (ViewRootImpl.java:2199)
  at android.view.ViewRootImpl$TraversalRunnable.run (ViewRootImpl.java:9088)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:999)
  at android.view.Choreographer.doCallbacks (Choreographer.java:797)
  at android.view.Choreographer.doFrame (Choreographer.java:732)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:984)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8154)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
```

**Comment by eddiemuc** [comment]
Looking into code on release, it seems that @moving-bits already introduced a fix/safeguard for this on 14.11.20:
![image](https://user-images.githubusercontent.com/6909759/99908978-069d3b80-2ce6-11eb-9a78-eb83c6513dea.png)
